controllers/argocd/util.go

```diff
@@ -1879,3 +1874,168 @@ func appendUniqueArgs(cmd []string, extraArgs []string) []string {
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldNSMgmt, okOld := e.ObjectOld.(*argoproj.NamespaceManagement)
+			newNSMgmt, okNew := e.ObjectNew.(*argoproj.NamespaceManagement)
+
+			if !okOld || !okNew {
+				return false
+			}
+
+			k8sClient, err := initK8sClient()
+			if err != nil {
+				log.Error(err, "Failed to initialize Kubernetes client")
+				return false
+			}
+
+			// If `.spec.managedBy` changes, trigger reconciliation
+			if oldNSMgmt.Spec.ManagedBy != newNSMgmt.Spec.ManagedBy {
+				// Delete RBACs for the old namespace
+				if err := deleteRBACsForNamespace(e.ObjectOld.GetNamespace(), k8sClient); err != nil {
+					log.Error(err, fmt.Sprintf("failed to delete RBACs for namespace: %s", e.ObjectOld.GetNamespace()))
+				} else {
+					log.Info(fmt.Sprintf("Successfully1 removed the RBACs for namespace: %s", e.ObjectOld.GetNamespace()))
+				}
+
+				// Delete namespace from cluster secret of previously managing argocd instance
+				if err = deleteManagedNamespaceFromClusterSecret(oldNSMgmt.Spec.ManagedBy, e.ObjectOld.GetNamespace(), k8sClient); err != nil {
+					log.Error(err, fmt.Sprintf("unable to delete namespace %s from cluster secret", e.ObjectOld.GetNamespace()))
+				} else {
+					log.Info(fmt.Sprintf("Successfully12 deleted namespace %s from cluster secret", e.ObjectOld.GetNamespace()))
+				}
+
+				// Return true to trigger reconciliation
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// If namespaceManagement CR is deleted, then delete RBACs for the namespace that was present in the NamespaceManagement CR.
+			nsMgmt, ok := e.Object.(*argoproj.NamespaceManagement)
+			if !ok {
+				return false
+			}
+
+			// Retrieve the ArgoCD instance that was managing this namespace
+			argocdNamespace := nsMgmt.Spec.ManagedBy
+			if argocdNamespace == "" {
+				// If .spec.managedBy is not set, there is no ArgoCD instance managing this namespace, so no cleanup is needed
+				log.Info("No ArgoCD CR specified in .spec.managedBy, skipping cleanup")
+				return false
+			}
+
+			k8sClient, err := initK8sClient()
+			if err != nil {
+				return false
+			}
+
+			if err := deleteRBACsForNamespace(e.Object.GetNamespace(), k8sClient); err != nil {
+				log.Error(err, fmt.Sprintf("failed to delete RBACs for namespace: %s", e.Object.GetNamespace()))
+			} else {
+				log.Info(fmt.Sprintf("Successfully removed the RBACs for namespace: %s", e.Object.GetNamespace()))
+			}
+
+			// Delete managed namespace from cluster secret
+			err = deleteManagedNamespaceFromClusterSecret(argocdNamespace, e.Object.GetNamespace(), k8sClient)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("unable to delete namespace %s from cluster secret", e.Object.GetNamespace()))
+			} else {
+				log.Info(fmt.Sprintf("Successfully deleted namespace %s from cluster secret", e.Object.GetNamespace()))
+			}
+			return false
+		},
+	}
+}
+
+func argoCDNamespaceManagementFilterPredicate() predicate.Predicate {
```

Comment:
Need unit tests for the logic defined in `argoCDNamespaceManagementFilterPredicates`. 

In order to test the functions, you may need to move the functions outside of `argoCDNamespaceManagementFilterPredicates` and into their own functions (which you can then reference from ``argoCDNamespaceManagementFilterPredicates`.`)